api/environments/identities/traits/views.py

```diff
@@ -33,21 +33,50 @@
     SDKCreateUpdateTraitSerializer,
 )
 from environments.views import logger
+from sse.decorators import generate_identity_update_message
+from sse.tasks import send_identity_update_messages
 from util.views import SDKAPIView
 
+generate_identity_message_decorator_trait_view = generate_identity_update_message(
+    lambda req: (req.environment, req.identity.identifier)
+)
+
 
+@method_decorator(
+    generate_identity_message_decorator_trait_view,
+    name="create",
+)
+@method_decorator(
+    generate_identity_message_decorator_trait_view,
+    name="destroy",
+)
+@method_decorator(
+    generate_identity_message_decorator_trait_view,
+    name="update",
+)
 class TraitViewSet(viewsets.ModelViewSet):
     serializer_class = TraitSerializer
 
+    def initial(self, request, *args, **kwargs):
+        # Add environment and identity to request(used by generate_identity_update_message decorator)
+        identity = Identity.objects.select_related("environment").get(
+            pk=self.kwargs["identity_pk"]
+        )
+        if not identity.environment.api_key == self.kwargs["environment_api_key"]:
+            raise Identity.DoesNotExists()
```

Comment:
I think this should be: 

```suggestion
            raise Identity.DoesNotExist()
```